2 - ListsLogicAndIterations/Code/Dictionarys.cs

```diff
@@ -3,21 +3,41 @@
 
 namespace ListsLogicAndIterations.Code
 {
-	public static class Dictionarys
-	{
+    public static class Dictionarys
+    {
 		public static Dictionary<Guid, decimal> Build(Guid key, decimal value)
 		{
-			throw new NotImplementedException();
+            Dictionary<Guid, decimal> result = new Dictionary<Guid, decimal>
+            {
+                { key, value }
+            };
+            return result;
 		}
 
 		public static decimal GetValue(Dictionary<Guid, decimal> dict, Guid key)
 		{
-			throw new NotImplementedException();
-		}
+            if (dict.TryGetValue(key, out _))
+            {
+                return dict[key];
+            }
+            else
+            {
+                Console.WriteLine("Value does not exist in the Dictionary");
+                return 0;
+            }
+        }
+       
 
 		public static void AddValue(Dictionary<Guid, decimal> dict, Guid key, decimal value)
 		{
-			throw new NotImplementedException();
+            try
+            {
+                dict.Add(key, value);
+            }
+            catch
+            {
+                Console.WriteLine("Value already exists in the Dictionary");
+            }
```

Comment:
Though this works, a try & catch is "expensive" and means you don't have control of your code. You can check to see if it contains the key already by doing:

```
			if(!dict.ContainsKey(key))
			{
				dict.Add(key, value);
			}
```